Automattic/vip-design-system
# New components: Menu & Breadcrumbs

## Description

This component is the base for the Responsiveness Project Header Part. I will create at least two more components:

- (THIS ONE) ~Navigation sidebar (Migration from the VIP Dashboard to a generic component)~
- Breadcrumbs (Breadcrumbs with some responsive behavior)
- Toolbar.Toggler (to create a hamburger menu and open the Navigation sidebar)

This Pull Request adds another variant of the `Nav` component. It's called Menu. This variant uses the same structure as the Nav component but adds support for vertical sub-menus, similar to the existing menus of the VIP Dashboard.


### What's included

- A base structure similar to the existing sidebar of the VIP Dashboard;
- Support for sub-menus;
- AddonHeader support (header between menu groups)
- Separators

### Screenshots

<img width="1068" alt="image" src="https://github.com/Automattic/vip-design-system/assets/3402/6d0ebfb1-6858-4445-961e-718fbc83da5b">

<img width="1041" alt="image" src="https://github.com/Automattic/vip-design-system/assets/3402/6f595643-0687-4c62-ba12-45e9f31a2e76">

### The Plan

After merging this component, I will put it in the VIP Dashboard. Once the other parts are done, we will move the other parts to the VIP Dashboard, including the Responsiveness behavior.

## Checklist


## Steps to Test — Menu

1. Open https://deploy-preview-313--vip-design-system-components.netlify.app/?path=/story/navigation-nav--menu
2. Navigate using the Keyboard
3. You can open/close
4. Items are announced as expected

## Steps to Test — Breadcrumbs

1. Open https://deploy-preview-313--vip-design-system-components.netlify.app/?path=/docs/navigation-breadcrumbs--docs
2. Navigate using the Keyboard
3. Last item is not a link
4. You can inspect the code and see a `nav`, and the last `li` with a `aria-current`
5. Use your browser to reduce the breakpoint to less than 654px, you should only see the penultimate item
